src/proto/single/resp.rs

```diff
@@ -118,6 +120,92 @@ pub async fn read_ok<R: AsyncBufRead + Unpin>(r: R) -> Result<(), Error> {
     Err(bad("server ok", &rr).into())
 }
 
+// ----------------------------------------------
+
+/// Faktory service stats.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[non_exhaustive]
+pub struct DataSnapshot {
+    /// Total number of job failures.
+    pub total_failures: u64,
+
+    /// Total number of processed jobs.
+    pub total_processed: u64,
+
+    /// Total number of enqueued jobs.
+    pub total_enqueued: u64,
+
+    /// Total number of queues.
+    pub total_queues: u64,
+
+    /// Queues stats.
+    ///
+    /// A mapping between a queue name and its size (number of jobs on the queue).
+    /// The keys of this map effectively make up a list of queues that are currently
+    /// registered in the Faktory service.
+    pub queues: HashMap<String, u64>,
+
+    /// ***Deprecated***. Faktory's task runner stats.
+    ///
+    /// Note that this is exposed as a "generic" `serde_json::Value` since this info
+    /// belongs to deep implementation details of the Faktory service.
+    #[deprecated]
+    pub tasks: serde_json::Value,
+}
+
+/// Faktory's server process stats.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ServerSnapshot {
+    /// Faktory's description (e.g. "Faktory").
+    pub description: String,
+
+    /// Faktory's version as semver (e.g. "1.8.0").
+    #[serde(rename = "faktory_version")]
+    pub version: String,
+
+    /// Faktory server process uptime in seconds.
+    pub uptime: u64,
```

Comment:
Can we expose this as `std::time::Duration` instead so units aren't going to confuse people (i.e., by using `#[serde(deserialize_with = )]` to construct it from seconds)?

Reply:
Done. Also bebug output would be
```
server: ServerSnapshot {
        description: "Faktory",
        version: Version {
            major: 1,
            minor: 8,
            patch: 0,
        },
        uptime: 6s,
        connections: 2,
        command_count: 1,
        used_memory_mb: 7,
    },
```
This `6s` fmt is just killing :smiley: